File: appcore/src/main/java/mobi/cangol/mobile/service/route/OnNavigation.java
package mobi.cangol.mobile.service.route;

import android.os.Bundle;

import androidx.fragment.app.Fragment;

/**
 * Created by xuewu.wei on 2018/11/2.
 */
public interface OnNavigation {

    void notFound(String path);

    void toFragment(Class<? extends Fragment> fragmentClass, Bundle bundle, boolean newStack);
}

File: appcore/src/main/java/mobi/cangol/mobile/service/route/OnNavigation.kt
package mobi.cangol.mobile.service.route

import android.content.Intent
import android.support.v4.app.Fragment

/**
 * Created by xuewu.wei on 2018/11/2.
 */
interface OnNavigation {

    fun toActivity(intent: Intent)

    fun toFragment(fragment: Fragment)
}
